### ISSUE TITLE
Stencil failing

### ISSUE THREAD

**Issue by jaeyoung-p** [wlruys/task4feedback#12]
```shell
python test_stencil.py -e 1 -m random  -o random -g 4 -pb 10 -dd 4 -d random
``` 
```shell
Traceback (most recent call last):
  File "/Users/jaeyoung/work/task4feedback/test/graphs/test_stencil.py", line 357, in <module>
    test_data()
  File "/Users/jaeyoung/work/task4feedback/test/graphs/test_stencil.py", line 253, in test_data
    simulated_time, task_order_log, success = simulator.run()
                                              ^^^^^^^^^^^^^^^
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/simulator.py", line 273, in run
    new_events = self.process_event(event)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/simulator.py", line 236, in process_event
    new_event_pairs = self.mechanisms[event](
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/schedulers/architecture.py", line 51, in wrapper
    return function(scheduler_state, event, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/schedulers/parla/architecture.py", line 845, in launcher
    task.notify_state(TaskState.LAUNCHED, objects.taskmap, current_time)
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/task.py", line 310, in notify_state
    if new_status := task.counters.notified_state(state):
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/task.py", line 129, in notified_state
    if self.check_count(new_state):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jaeyoung/miniconda3/lib/python3.12/site-packages/task4feedback/simulator/task.py", line 119, in check_count
    assert value >= 0
AssertionError
``` 

can be reproducible by running script in test/graphs rl-branch

**Comment by jaeyoung-p** [comment]
```shell
python test_stencil.py -e 1 -m heft -o heft -g 4 -pb 10 -dd 4 -d rr
python test_stencil.py -e 1 -m heft -o heft -g 4 -pb 10 -dd 4 -d random
python test_stencil.py -e 1 -m random  -o heft -g 4 -pb 10 -dd 4 -d random
python test_stencil.py -e 1 -m eft_without_data  -o heft -g 8 -pb 10 -dd 4 -d rr
python test_stencil.py -e 1 -m eft_without_data  -o random -g 8 -pb 10 -dd 4 -d rr
```

all fails with the same error

**Comment by wlruys** [comment]
Yep, something is duplicating dependencies.

**Comment by wlruys** [comment]
Something _was_ duplicating dependencies (error in bfs search), that' s now fixed but it wasn't causing this issue.

**Comment by wlruys** [comment]
Caused by int8 used in dependency counter. 

Fixed in c725649940b5f867fe745ae58f9fc509225fafbf along with many other minor related bugs (removing duplicated dependencies and fixing status counters for data tasks), that while didn't effect correctness made tracking this down harder..
